### ISSUE TITLE
Use lockedEnvironment in newSCESet

### ISSUE THREAD

**Issue by LTLA** [davismcc/archive-scater#47]
Behold the weird and wondrous behaviour when environments are passed by reference:

``` r
require(scater)
example(newSCESet)
copy <- example_sceset

exprs(copy)[1,] <- 1
exprs(copy)[1,] # fine
exprs(example_sceset)[1,] # unexpected change to original object
```

It'd be better to switch to `'lockedEnvironment'` in the `assayDataNew` call in `newSCESet`.

**Comment by LTLA** [comment]
Note that, as a result, bindings cannot be added directly, but has to go through:

``` r
assayDataElement(example_sceset, "Y") <- Y # or whatever you want to add
```

**Comment by davismcc** [comment]
Thanks, Aaron. Fixed in v1.1.7, commit c46b127 .
